soul-plugin/soul-plugin-ratelimiter/src/main/java/org/dromara/soul/plugin/ratelimiter/RateLimiterPlugin.java

```diff
@@ -61,18 +64,26 @@ public int getOrder() {
         return PluginEnum.RATE_LIMITER.getCode();
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     protected Mono<Void> doExecute(final ServerWebExchange exchange, final SoulPluginChain chain, final SelectorData selector, final RuleData rule) {
         RateLimiterHandle limiterHandle = RatelimiterRuleHandleCache.getInstance()
                 .obtainHandle(RateLimiterPluginDataHandler.getCacheKeyName(rule));
-        return redisRateLimiter.isAllowed(rule.getId(), limiterHandle)
+        return redisRateLimiter.isAllowed(exchange, rule.getId(), limiterHandle)
                 .flatMap(response -> {
                     if (!response.isAllowed()) {
                         exchange.getResponse().setStatusCode(HttpStatus.TOO_MANY_REQUESTS);
                         Object error = SoulResultWrap.error(SoulResultEnum.TOO_MANY_REQUESTS.getCode(), SoulResultEnum.TOO_MANY_REQUESTS.getMsg(), null);
                         return WebFluxResultUtils.result(exchange, error);
                     }
                     return chain.execute(exchange);
+                }).doFinally(s -> {
+                    RateLimiterAlgorithm<?> rateLimiterAlgorithm = (RateLimiterAlgorithm<?>) exchange.getAttributes().get(RedisRateLimiter.RATE_LIMITER_ALGORITHM);
+                    if (rateLimiterAlgorithm != null) {
+                        List<String> keys = (List<String>) exchange.getAttributes().get(RedisRateLimiter.KEYS);
+                        List<String> scriptArgs = (List<String>) exchange.getAttributes().get(RedisRateLimiter.SCRIPT_ARGS);
+                        rateLimiterAlgorithm.callback(null, keys, scriptArgs);
```

Comment:
Do I need to change the parameters of the callback method？because I don't think scripts will be passed to the method